llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -2861,6 +2863,19 @@ static SDValue lowerCTLZ_CTTZ_ZERO_UNDEF(SDValue Op, SelectionDAG &DAG) {
   return DAG.getNode(ISD::SUB, DL, VT, DAG.getConstant(Adjust, DL, VT), Trunc);
 }
 
+/// Emit a replacement for the (to-be) removed CFromPtr instruction.
+static SDValue
+emitCFromPtrReplacement(SelectionDAG &DAG, const SDLoc &DL, SDValue Base,
+                        SDValue IntVal, EVT CLenVT,
+                        unsigned Intrin = Intrinsic::cheri_cap_offset_set) {
```

Comment:
Same as CToPtr, we should be consistent about offset vs address

Reply:
As above, if someone explicitly wrote __builtin_cheri_cap_from_ptr (or the assembly) we probably shouldn't change the meaning of that?